airbyte-integrations/csv-destination/src/test-integration/java/io/airbyte/integrations/destination/csv/CsvDestinationIntegrationTest.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2020 Airbyte
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.airbyte.integrations.destination.csv;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.integrations.base.TestDestination;
+import java.io.FileReader;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVRecord;
+
+public class CsvDestinationIntegrationTest extends TestDestination {
+
+  private static final String COLUMN_NAME = "data";
+  private static final Path RELATIVE_PATH = Path.of("integration_test/test");
+
+  @Override
+  protected String getImageName() {
+    return "airbyte/airbyte-csv-destination:dev";
+  }
+
+  @Override
+  protected JsonNode getConfig() {
+    return new ObjectMapper().createObjectNode().put("destination_path", Path.of("/local").resolve(RELATIVE_PATH).toString());
+  }
+
+  @Override
+  protected JsonNode getInvalidConfig() {
+    return new ObjectMapper().createObjectNode().put("destination_path", Path.of("/@").resolve(RELATIVE_PATH).toString());
+  }
+
+  @Override
+  protected List<JsonNode> recordRetriever(TestDestinationEnv testEnv) throws Exception {
+    final List<Path> list = Files.list(testEnv.getLocalRoot().resolve(RELATIVE_PATH)).collect(Collectors.toList());
+    assertEquals(1, list.size());
```

Comment:
It's unintuitive that a `retriever` method does assertions, but I'm not sure I have a much better suggestion. Assuming "destination-specific" tests like number of files in directory are always run after a sync, would it make sense to have a separate method for validations than for retrieving?

Reply:
i'll remove the assertion. i agree. it doesn't belong here.

Reply:
put a comment to add a get tables abstract method that we can use this check. going to leave assertion here for now.